### DISCUSSION TITLE
Please help me use JSONata to remove duplicate content.

### DISCUSSION THREAD

**Discussion by seasoncool** [grafana/grafana-infinity-datasource#707] [Q&A] [answered]
The following is my JSON response. 

[
  {"id": "45018", "State": "1 - Critical"},
  {"id": "45018", "State": "1 - Critical"},
  {"id": "45018", "State": "1 - Critical"}
]
I want to remove duplicate combinations of id and State values. 

My expected result is:

[
  {"id": "45018", "State": "1 - Critical"}
]

How can I use JSONata expressions to achieve this? thanks.

**Comment by aakashH242** [comment, marked as answer]
You can use the [distinct](https://docs.jsonata.org/array-functions#distinct) function of JSONata to remove duplicates.
Don't forget to set the parser to UQL.
Please try the following query:
```
parse-json
| jsonata "$distinct($)"
```

![image](https://github.com/grafana/grafana-infinity-datasource/assets/20237354/fdda4ab3-ebc2-4aeb-ba21-caf89aab4e30)

**Comment by seasoncool** [comment]
thanks.
